src/HMRC.ESFA.Levy.Api.UnitTests/PaymentStatusesProcessorTests.cs

```diff
@@ -0,0 +1,241 @@
+
+            var dateAccountCreatedStartOfYear = new DateTime(2017, 01, 01, 00, 00, 00, DateTimeKind.Utc);
+
+            var declarationsAccountCreatedAtStartOfYear = GetDeclarationList();
+
+            _declarationsAccountCreatedAtStartOfYearPostProcessed = _processor
+                .ProcessDeclarationPaymentStatuses(declarationsAccountCreatedAtStartOfYear,
+                    dateAccountCreatedStartOfYear);
+
+        }
+
+        private static List<Declaration> GetDeclarationList()
+        {
+            return new List<Declaration>
+            {
+                new Declaration
+                {
+                    Id = "Late2",
+                    SubmissionTime = new DateTime(2017, 12, 20, 00, 00, 00, DateTimeKind.Utc),
+                    PayrollPeriod = new PayrollPeriod {Month = 5, Year = PayrollYear},
+                },
+                new Declaration
+                {
+                    Id = "unprocessedAndVeryEarly",
+                    SubmissionTime = new DateTime(2017, 1, 20, 00, 00, 00, DateTimeKind.Utc),
+                    PayrollPeriod = new PayrollPeriod {Month = 5, Year = PayrollYear},
+                },
+                new Declaration
+                {
+                    Id = "LastBeforeCutoff",
+                    SubmissionTime = new DateTime(2017, 09, 19, 23, 59, 59, DateTimeKind.Utc),
+                    PayrollPeriod = new PayrollPeriod {Month = 5, Year = PayrollYear}
+                },
+                new Declaration
+                {
+                    Id = "secondLastBeforeCutoff",
+                    SubmissionTime = new DateTime(2017, 09, 19, 10, 15, 00, DateTimeKind.Utc),
+                    PayrollPeriod = new PayrollPeriod {Month = 5, Year = PayrollYear},
+                },
+                new Declaration
+                {
+                    Id = "Late1",
+                    SubmissionTime = new DateTime(2017, 09, 20, 0, 0, 0, DateTimeKind.Utc),
+                    PayrollPeriod = new PayrollPeriod {Month = 5, Year = PayrollYear},
+                },
+                new Declaration
+                {
+                    Id = "early",
+                    SubmissionTime = new DateTime(2017, 8, 19, 00, 00, 00, DateTimeKind.Utc),
+                    PayrollPeriod = new PayrollPeriod {Month = 4, Year = PayrollYear},
+                },
+                new Declaration
+                {
+                    Id = "noPayrollPeriod",
+                    SubmissionTime = new DateTime(2017, 8, 19, 00, 00, 00, DateTimeKind.Utc)
+                },
+            };
+        }
+
+        [Test]
+        public void ShouldReturnEmptyListIfPassedEmptyList()
+        {
+            Assert.AreEqual(_emptyDeclarationPostProcessed.Count, 0);
+        }
+
+        [Test]
+        public void ShouldSetLate1PaymentStatusToLate()
+        {
+            var expectedLateEntry = _declarationsPostProcessed.First(x => x.Id == "Late1");
```

Comment:
There's a potential for a typo by selecting based on a string here. you could use statics for the test data

```c#
                private static Declaration noPayrollPeriod = new Declaration
                {
                     Id = "noPayrollPeriod",
                     SubmissionTime = new DateTime(2017, 8, 19, 00, 00, 00, DateTimeKind.Utc)
                 };
```